### ISSUE TITLE
New Queues, Exchanges, Bindings not been created.

### ISSUE THREAD

**Issue by ocristian** [freenowtech/spring-multirabbit#54]
Queues, exchanges or bindings not existing in the broker yet are not being automatically created when using `@RabbitListener`.

A possible root cause for it is because we have specific `declaringAdmins`, in those declarebles created by MultiRabbit.
[This info is evaluated by RabbitAdmin while filtering the declarables.](https://github.com/spring-projects/spring-amqp/blob/70d79b5ba540932cf8ae765dd05174c427019e26/spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitAdmin.java#L643)

Here, we can see the obj status during that evaluation. 
Queue `declaringAdmins` is different than `this.beanName`

```
dec = {Queue@11016} "Queue [name=new-queue-name, durable=true, autoDelete=false, exclusive=false, arguments={},   actualName=new-queue-name]"  
 name = "new-queue-name"   
 durable = true  
 exclusive = false  
 autoDelete = false    
 actualName = "new-queue-name"  
 shouldDeclare = true  
 declaringAdmins = {ArrayList@11026}  size = 1  
  0 = "eventbus-admin" 
 ignoreDeclarationExceptions = true  
 arguments = {HashMap@11027}  size = 0  
this.explicitDeclarationsOnly = false  
this.beanName = "amqpAdmin"  
```

**Comment by rwanderc** [comment]
Thank @ocristian for opening the issue. That's fixed in `main` and will be merged and released in 2.4.0.

**Comment by rwanderc** [comment]
btw @ocristian can you pls test with latest `main` again?

**Comment by ocristian** [comment]
Issue fixed on release v2.4.0.
